src/algorithm/good_thomas_algorithm.rs

```diff
@@ -9,26 +11,22 @@ use algorithm::FFTAlgorithm;
 pub struct GoodThomasAlgorithm<T> {
     width: usize,
     // width_inverse: usize,
-    width_size_fft: Box<FFTAlgorithm<T>>,
+    width_size_fft: Rc<FFTAlgorithm<T>>,
 
     height: usize,
     // height_inverse: usize,
-    height_size_fft: Box<FFTAlgorithm<T>>,
+    height_size_fft: Rc<FFTAlgorithm<T>>,
```

Comment:
At some point I'd like to enable this library to work without the standard library. Is this heap allocation necessary, or could we get away with using generics? Something like `struct GoodThomasAlgorithm<T, A> where A: FFTAlgorithm<T>`.